packages/trader/src/RangeTrader.ts

```diff
@@ -141,7 +141,7 @@ export class RangeTrader {
     }
 
     // Wait exactly one block to fetch events. This ensures that the events have been indexed by your node.
-    await blockUntilBlockMined(this.web3, (await this.web3.eth.getBlockNumber()) + 1);
+    await blockUntilBlockMined(this.web3, tradeExecutionTransaction.blockNumber);
```

Comment:
am i missing something here, but if the `tradeExecutionTransaction` has a `blockNumber` on it, doesnt that mean its been mined and you dont need to wait?

Reply:
I want @chrismaree to review this change specifically. I don't know exactly what the original intention of this was. It might've been to add a 1 block buffer after _one_ node said that it was mined because you may get load balanced to another node that may be a little behind. In that case, my solution isn't sufficient to do what @chrismaree was originally intending.

However, to answer your question @daywiss, it does mean that you've passed that block, but it doesn't mean that all infura nodes think you've passed that block. You could get load balanced to a different one. This causes issues in testing, though, because blocks are _only_ mined by hardhat when transactions are submitted (I have no idea how this ever worked in test environments).

@chrismaree may suggest that we find another workaround, which may mean that we briefly turn on scheduled mining rather than automining around these transactions in the test.

Reply:
yup @mrice32, you are exactly right as to why this bit of code was here. What would happen in production (only some times) is the bot would execute a trade and then it would report on the trade and the logic below would fetch the state on-chain regarding the pool amounts but it would not yet have captured the trade for some or other reason. Load balancing and some kind of delay in Infura makes the most sense. the simplest solution was to just force the bot to wait ~13 seconds at which point the nodes are "up to date" and then the logging works as expected.

what you have implemented here I don't think will achieve this effect correctly and the logs will still be invalid. can you not just simply change it to be:

```js
await blockUntilBlockMined(this.web3,tradeExecutionTransaction.blockNumber + 1);

```

how it worked in tests, like you pointed out is a good question. offhand I can remember how I dealt with this but glad it's working now!

Reply:
i see ok that makes sense. thanks for the context

Reply:
Thanks for the context @chrismaree. I will correct this and get your review on it next week!